### PR TITLE
Use per-env Auth0 accounts

### PR DESCRIPTION
**Pull request by andyhd**
# What

Use per-environment Auth0 credentials. Standardised credentials names across jenkins CI and control panel jenkins.

### CODE REVIEW

File: vars/deploy.groovy [hunk 1]
```diff
@@ -59,10 +58,9 @@ def call() {
           --set app.baseHost=${env.APP_BASE_DOMAIN} \
           --set shinyApp.docker.repository=${env.DOCKER_REGISTRY}/${env.REPO_NAME} \
           --set shinyApp.docker.tag=${env.DOCKER_TAG} \
-          --set OAuth.ClientSecret=${env.AUTH0_CLIENT_SECRET} \
-          --set OAuth.ClientId=${env.AUTH0_CLIENT_ID} \
+          --set OAuth.ClientSecret=${env.CLIENT_SECRET} \
```
Comment (xoen) [inline vars/deploy.groovy]: I didn't mind the `AUTH0_` prefix as it made very explicit these were coming from Auth0 (as the domain which still has the prefix).

File: vars/withOAuthClient.groovy [hunk 2]
```diff
@@ -1,18 +1,16 @@
 def call(Closure body) {
 
     env.NAME = "${env.RELEASE}.${env.APP_BASE_DOMAIN}"
+    env.AUTHZ_API_ID = "urn:auth0-authz-api"
```
Comment (xoen) [inline vars/withOAuthClient.groovy]: This is because it's always constant for the Authorisation extension API right?

**Comment by xoen** [comment]
LGTM

File: vars/withOAuthClient.groovy (hunk 2 above)
Comment (andyhd) [inline vars/withOAuthClient.groovy]: Yep

File: vars/deploy.groovy (hunk 1 above)
Comment (andyhd) [inline vars/deploy.groovy]: These are actually different Auth0 credentials, but I think they need better names to be clearer.

File: vars/deploy.groovy [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 
-  env.AUTH0_DOMAIN = 'kerinmoj.eu.auth0.com'
-  env.AUDIENCE = 'https://kerinmoj.eu.auth0.com/api/v2/'
+  env.AUTH0_DOMAIN = '${env.ENV}-analytics-moj.eu.auth0.com'
+  env.AUDIENCE = 'https://${env.AUTH0_DOMAIN}/api/v2/'
   env.REPO_NAME = env.JOB_NAME.split('/')[1]
   def commit = sh(script: 'git rev-parse HEAD', returnStdout: true)
   env.DOCKER_TAG = commit.substring(1, 8)
   env.DOCKER_REGISTRY = '593291632749.dkr.ecr.eu-west-1.amazonaws.com'
-  env.COOKIE_SECRET = 'XXXChangeThisXXX'
```
Comment (kerin) [inline vars/deploy.groovy]: How is cookie secret being handled now? In a Helm values YML file?

Comment (andyhd) [inline vars/deploy.groovy]: Yeah, I'm going to put it in secrets.yaml

**Pull request merged**